modin/pandas/test/dataframe/test_reduction.py

```diff
@@ -361,3 +362,30 @@ def test_reduction_specific(fn, numeric_only, axis):
         *create_test_dfs(test_data_diff_dtype),
         lambda df: getattr(df, fn)(numeric_only=numeric_only, axis=axis),
     )
+
+
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("subset_len", [1, 2])
+def test_value_counts(subset_len, sort):
+    data = test_data_values[0]
+    md_df, pd_df = create_test_dfs(data)
+    subset = [pd_df.columns[i * ((-1) ** i)] for i in range(subset_len)]
+
+    md_res = md_df.value_counts(subset=subset, sort=sort)
+    pd_res = pd_df.value_counts(subset=subset, sort=sort)
+
+    if subset_len == 1:
+        # 'pandas.DataFrame.value_counts' always returns frames with MultiIndex,
```

Comment:
We should add a note to the docs on this.

Reply:
answered [here](https://github.com/modin-project/modin/pull/2730#discussion_r685169024)

Reply:
Can you elaborate why the behavior is expensive to mimic?

Reply:
to mimic this behavior we have to convert regular `Index` to a `MultiIndex` at the end of `DataFrame.value_counts`, this is expensive not only because we would need to trigger the index computation (critical for OmniSci backend with its lazy execution) but because of the actual creation of MultiIndex object which may block execution for a while in the case of really huge Index axis (we use pandas versions of Index objects, so no parallelism here).